src/rimage.c

```diff
@@ -115,19 +115,21 @@ int main(int argc, char *argv[])
 		return -EINVAL;
 	}
 
-	/* firmware version and build id */
+	/* firmware version that can be major.minor or major.minor.micro */
 	if (image.fw_ver_string) {
-		ret = sscanf(image.fw_ver_string, "%hu.%hu",
+		ret = sscanf(image.fw_ver_string, "%hu.%hu.%hu",
 			     &image.fw_ver_major,
-			     &image.fw_ver_minor);
+			     &image.fw_ver_minor,
+			     &image.fw_ver_micro);
 
-		if (ret != 2) {
+		if (ret != 2 && ret != 3) {
```

Comment:
The return value of `sscanf` is the number of receiving arguments successfully assigned or EOF, I don't think we should allow partial assignment. the ret must be 3

Reply:
@aiChaoSONG Thanks for your comments. This is to be compatible with current SOF CMake that only passes 2 version arguments (major.minor) to rimage. If this rimage PR gets merged at first,  SOF building will fail as CMake doesn't pass 3 version arguments (major.minor.micro) as rimage expects. 

If SOF CMake https://github.com/thesofproject/sof/pull/5760 can also be merged, I can update rimage to only support 3 version arguments.

Reply:
Even the PR get merged, it will not be used, rimage is a submodule of SOF, unless rimage is upgraded on SOF side, new changes will never be used. so it's safe.

Reply:
I see. Thanks! I Will update PR after Rander comments if this PR will impact Windows.

Reply:
Fixed this in new PR#92 @aiChaoSONG